File: tests/code_examples/ruby/simple_class.rb
# Simple Ruby class for testing basic functionality.
#
# This file contains basic Ruby constructs to test
# GraphBuilder's ability to parse and analyze Ruby code.

class SimpleClass
  attr_reader :value
  attr_writer :name
  attr_accessor :status
  
  def initialize(value)
    @value = value
    @name = nil
    @status = 'initialized'
  end
  
  def get_value
    @value
  end
  
  def set_value(new_value)
    @value = new_value
  end
  
  def process_value
    processed = internal_process
    processed
  end
  
  private
  
  def internal_process
    "Processed: #{@value}"
  end
end

def simple_function
  'Hello from Ruby'
end

def function_with_parameter(name)
  "Hello, #{name}!"
end

def function_with_block
  yield if block_given?
end

def function_with_multiple_params(name, age = 25, *args, **kwargs)
  result = "Name: #{name}, Age: #{age}"
  result += ", Args: #{args}" unless args.empty?
  result += ", Kwargs: #{kwargs}" unless kwargs.empty?
  result
end

# Module-level constants
MODULE_CONSTANT = 'test_constant'
DEFAULT_VALUE = 42

# Module-level variables
$global_variable = 'global'
@@class_variable = 'class_variable'

# Usage examples
instance = SimpleClass.new('test_value')
result = simple_function
processed = instance.process_value

# Method with different parameter types
def demonstrate_ruby_features(required, optional = 'default', *splat, keyword:, optional_keyword: nil, **double_splat, &block)
  {
    required: required,
    optional: optional,
    splat: splat,
    keyword: keyword,
    optional_keyword: optional_keyword,
    double_splat: double_splat,
    has_block: block_given?
  }
end

File: tests/code_examples/typescript/simple_class.ts
/**
 * Simple TypeScript class for testing basic functionality.
 * 
 * This file contains basic TypeScript constructs to test
 * GraphBuilder's ability to parse and analyze TypeScript code.
 */

export interface ValueHolder {
  getValue(): string;
  setValue(value: string): void;
}

export class SimpleClass implements ValueHolder {
  private value: string;
  
  constructor(value: string) {
    this.value = value;
  }
  
  public getValue(): string {
    return this.value;
  }
  
  public setValue(value: string): void {
    this.value = value;
  }
  
  public processValue(): string {
    const processed = this.internalProcess();
    return processed;
  }
  
  private internalProcess(): string {
    return `Processed: ${this.value}`;
  }
}

export function simpleFunction(): string {
  return "Hello from TypeScript";
}

export function functionWithParameter(name: string): string {
  return `Hello, ${name}!`;
}

export function genericFunction<T>(item: T): T {
  return item;
}

// Type alias
export type ProcessorResult = {
  success: boolean;
  data: string;
  timestamp: Date;
};

// Enum
export enum Status {
  PENDING = "pending",
  PROCESSING = "processing", 
  COMPLETED = "completed",
  FAILED = "failed"
}

// Module-level constants
export const MODULE_CONSTANT = "test_constant";
export const DEFAULT_STATUS = Status.PENDING;

// Module-level usage
const instance = new SimpleClass("test");
const result = simpleFunction();